specification/extendedlocation/resource-manager/Microsoft.ExtendedLocation/preview/2021-08-31-preview/customlocations.json

```diff
@@ -0,0 +1,1196 @@
+      "type": "object",
+      "x-ms-azure-resource": true,
+      "x-ms-client-flatten": true,
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/resourceSyncRuleProperties",
+          "description": "The Resource Sync Rules patchable properties.",
+          "x-ms-client-flatten": true
+        },
+        "tags": {
+          "description": "Resource tags",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "resourceSyncRule": {
+      "description": "Resource Sync Rules definition.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/resourceSyncRuleProperties",
+          "description": "The set of properties specific to a Resource Sync Rule",
+          "x-ms-client-flatten": true
+        },
+        "systemData": {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData",
+          "description": "Metadata pertaining to creation and last modification of the resource",
+          "readOnly": true
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/TrackedResource"
+        }
+      ]
+    },
+    "resourceSyncRuleListResult": {
+      "description": "The List Resource Sync Rules operation response.",
+      "type": "object",
+      "properties": {
+        "nextLink": {
+          "description": "The URL to use for getting the next set of results.",
+          "type": "string",
+          "readOnly": true
+        },
+        "value": {
+          "description": "The list of Resource Sync Rules.",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/resourceSyncRule"
+          }
+        }
+      }
+    },
+    "resourceSyncRuleProperties": {
+      "description": "Properties for a resource sync rule.",
+      "type": "object",
+      "properties": {
+        "priority": {
+          "description": "Priority represents a priority of the Resource Sync Rule",
+          "type": "integer",
+          "format": "int32"
+        },
+        "provisioningState": {
+          "description": "Provisioning State for the Resource Sync Rule.",
+          "type": "string",
+          "readOnly": true
+        },
+        "selector": {
+          "description": "A label selector is a label query over a set of resources. The result of match labels and match expressions should both be satisfied.",
+          "type": "object",
+          "properties": {
+            "matchExpressions": {
+              "$ref": "#/definitions/matchExpressionsList",
+              "description": "MatchExpressions contain values, a key, and an operator that relate the key and values.",
+              "x-ms-client-flatten": true
+            },
+            "matchLabels": {
```

Comment:
why are there both expressions and labels?  Seems like any label could be represented as an expression

Reply:
Yes, agreed. Actually this is the same concept from Kubernetes selector, we didn't make customized changes here.
https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#resources-that-support-set-based-requirements